src/sulogin-shell/sulogin-shell.c

```diff
@@ -103,17 +106,33 @@ int main(int argc, char *argv[]) {
                 /* allows passwordless logins if root account is locked. */
                 sulogin_cmdline[1] = "--force";
 
-        (void) fork_wait(sulogin_cmdline);
-
         r = bus_connect_system_systemd(&bus);
         if (r < 0) {
                 log_warning_errno(r, "Failed to get D-Bus connection: %m");
                 r = 0;
-        } else {
+        }
+
+        path = unit_dbus_path_from_name(SPECIAL_DEFAULT_TARGET);
+        if (!path)
+                return log_oom();
+
+        do {
+                (void) fork_wait(sulogin_cmdline);
+
                 (void) reload_manager(bus);
```

Comment:
`bus` may be NULL here. Please conditionalize the call or make `reload_manager()` noop when `bus` is `NULL`.

Reply:
I see. Looks like overlooked since 375c3f6aae.